Sources/DistributedActors/Pattern/WorkerPool.swift

```diff
@@ -44,19 +54,79 @@ public struct WorkerPool<Message: ActorMessage> {
         /// The worker pool will terminate itself if all of its static workers have terminated.
         /// You may death-watch the worker pool in order to react to this situation, e.g. by spawning a replacement pool,
         /// or gracefully shutting down your application.
-        case `static`([_ActorRef<Message>])
+        case `static`([Worker])
     }
 
     // ==== ------------------------------------------------------------------------------------------------------------
     // MARK: State shared across all states of the WorkerPool
 
-    var settings: WorkerPoolSettings<Message>
+    var settings: WorkerPoolSettings<Worker>
     var selector: Selector {
         self.settings.selector
     }
-
-    init(settings: WorkerPoolSettings<Message>) {
+    
+    /// The worker pool. Worker will be selected round-robin.
+    private var workers: [Weak<Worker>] = []
+    private var roundRobinPos = 0
+    
+    /// Boolean flag to help determine if pool becomes empty because at least one worker has terminated.
+    private var hasTerminatedWorkers = false
+
+    init(settings: WorkerPoolSettings<Worker>, system: ActorSystem) async {
         self.settings = settings
+        self.actorSystem = system
+
+        switch self.selector {
+        case .dynamic(let key):
+            Task {
+                for await worker in await self.actorSystem.receptionist.subscribe(to: key) {
+                    system.log.log(level: self.settings.logLevel, "Got listing member for \(self.selector): \(worker)")
+                    self.workers.append(Weak(worker))
+                    watchTermination(of: worker) { self.onWorkerTerminated(id: $0) }
+                }
+            }
+        case .static(let workers):
+            self.workers.append(contentsOf: workers.map(Weak.init))
+            workers.forEach { worker in
+                watchTermination(of: worker) { self.onWorkerTerminated(id: $0) }
+            }
+        }
+    }
+    
+    public distributed func submit(work: WorkItem) async throws -> WorkResult {
+        let worker = try await self.selectWorker()
+        return try await worker.submit(work: work)
+    }
+    
+    private func selectWorker() async throws -> Worker {
+        // Wait if we haven't received the initial workers listing yet.
+        // Otherwise, the pool has become empty because all workers have been terminated,
+        // in which case we either wait for new workers or throw error.
+        while self.workers.isEmpty {
+            switch (self.hasTerminatedWorkers, self.settings.whenAllWorkersTerminated) {
+            case (false, _), (true, .awaitNewWorkers):
+                try await Task.sleep(nanoseconds: 100)
+            case (true, .crash(let error)):
+                throw error
+            }
+        }
```

Comment:
Probably also some max timeout for making the call here, so we should stop after callTimeout perhaps and throw from here?